h2o-r/h2o-package/R/connection.R

```diff
@@ -190,8 +190,6 @@ h2o.init <- function(ip = "localhost", port = 54321, startH2O = TRUE, forceDL =
         print(rv$httpStatusCode)
         print(rv$curlErrorMessage)
 
-        #try a hail mary curl
```

Comment:
Why is this being removed? Doesn't seem related.

Reply:
I believe it *is* related, because part of the problem was a confusing message:
- in case of *any* error that prevents h2o from starting, it emphasises the false impression that the problem is with connection
- and when the problem really *is* just connection, it pollutes the console output with lines providing no additional information, because, in such a case, the connection problem is already mentioned right above curl output, look here:

```
> h2o.init()
Reading in config file: ./.h2oconfig

H2O is not running yet, starting it now...

Note:  In case of errors look at the following log files:
    /tmp/RtmpLuhky4/h2o_pk_started_from_r.out
    /tmp/RtmpLuhky4/h2o_pk_started_from_r.err

java version "1.6.0_45"
Java(TM) SE Runtime Environment (build 1.6.0_45-b06)
Java HotSpot(TM) 64-Bit Server VM (build 20.45-b01, mixed mode)

Starting H2O JVM and connecting: ............................................................ 
[1] "localhost"
[1] 54321
[1] TRUE
[1] -1
[1] "Failed to connect to localhost port 54321: Connection refused"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 54321: Connection refused
[1] 7
Error in h2o.init() : H2O failed to start, stopping execution.
```
(this console output was created on `master` branch)


And these lines are what we loose by removing that curl:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 54321: Connection refused
[1] 7
```

I don't see anything useful here.

Also this result can hardly be different, because it comes in situation when the previous timeout has expired, and the connection is unlikely to be working the very nanosecond after timeout.

I can imagine that, if R uses some other network configuration (of proxies etc), it might fail, while this curl goes directly (or based on system configuration) and in such a case it might behave differently.
But, if that was the purpose (I hope not, would be a bit weird), it should be elaborated into more comprehensive information for the user, otherwise it is really just confusing.

Reply:
In general it is a good idea to provide more information to debug the issue but in this case I think we can really remove it because:

- it hardcodes HTTP
- it relies on `curl` (useless on windows)
- it hardcodes 54321

This code is working. Let's remove it.